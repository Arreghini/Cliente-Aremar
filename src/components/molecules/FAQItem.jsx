import { useState } from 'react';
import PropTypes from 'prop-types';

const FAQItem = ({ question, answer }) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleOpen = () => setIsOpen(prev => !prev);

  return (
    <div className="w-full max-w-3xl min-w-[768px] mx-auto bg-white p-8 rounded-2xl shadow-md border border-mar-claro transition-all duration-300">
      <button
        onClick={toggleOpen}
        className="w-full text-left flex justify-between items-center font-semibold text-mar-profundo hover:text-playa-sol"
      >
        <span className="flex-1">{question}</span>
        <span aria-hidden="true" className="text-xl ml-4 w-6 text-center">
          {isOpen ? '-' : '+'}
        </span>
      </button>
      <div
        className="transition-all duration-300 ease-in-out overflow-hidden"
        style={{ maxHeight: isOpen ? '500px' : '0' }}
      >
        <p className="mt-2 text-neutral-oscuro font-body">{answer}</p>
      </div>
    </div>
  );
};

FAQItem.propTypes = {
  question: PropTypes.string.isRequired,
  answer: PropTypes.string.isRequired,
};

export default FAQItem;
