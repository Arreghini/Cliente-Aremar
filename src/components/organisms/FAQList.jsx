import React from 'react';
import FAQItem from '../molecules/FAQItem';

const faqs = [
  {
    question: '¿Cómo hago una reserva?',
    answer:
      'Podés hacer una reserva desde la página de inicio seleccionando fechas y tipo de habitación.',
  },
  {
    question: '¿Se puede cancelar una reserva?',
    answer:
      'Sí, podés cancelar desde tu perfil. Consultá nuestras políticas de cancelación.',
  },
  {
    question: '¿Aceptan mascotas?',
    answer:
      'En algunos departamentos sí, según disponibilidad. Consultá antes de reservar.',
  },
];

const FAQList = () => {
  return (
    <div className="space-y-4" data-testid="faq-list-container">
      {faqs.map((faq, index) => (
        <FAQItem key={index} question={faq.question} answer={faq.answer} />
      ))}
    </div>
  );
};

export default FAQList;
