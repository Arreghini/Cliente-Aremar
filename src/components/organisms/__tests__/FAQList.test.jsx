import { describe, expect, test, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import FAQList from '../FAQList';

// Mock de FAQItem
vi.mock('../../molecules/FAQItem', () => ({
  default: vi.fn(({ question, answer, onClick }) => (
    <div data-testid="faq-item" onClick={onClick}>
      <p>{question}</p>
      <p>{answer}</p>
    </div>
  )),
}));

import FAQItem from '../../molecules/FAQItem';

describe('FAQList', () => {
  test('renderiza el contenedor principal', () => {
    render(<FAQList />);
    const container = screen.getByTestId('faq-list-container');
    expect(container).toBeInTheDocument();
  });

  test('renderiza todos los FAQItems', () => {
    render(<FAQList />);
    const items = screen.getAllByTestId('faq-item');
    expect(items.length).toBe(3); // Debe coincidir con el array faqs
  });

  test('cada FAQItem recibe la pregunta y respuesta correcta', () => {
    render(<FAQList />);
    expect(FAQItem).toHaveBeenCalledWith(
      {
        question: '¿Cómo hago una reserva?',
        answer:
          'Podés hacer una reserva desde la página de inicio seleccionando fechas y tipo de habitación.',
      },
      {}
    );
    expect(FAQItem).toHaveBeenCalledWith(
      {
        question: '¿Se puede cancelar una reserva?',
        answer:
          'Sí, podés cancelar desde tu perfil. Consultá nuestras políticas de cancelación.',
      },
      {}
    );
    expect(FAQItem).toHaveBeenCalledWith(
      {
        question: '¿Aceptan mascotas?',
        answer:
          'En algunos departamentos sí, según disponibilidad. Consultá antes de reservar.',
      },
      {}
    );
  });

  test('cada FAQItem muestra la pregunta y respuesta en pantalla', () => {
    render(<FAQList />);
    expect(screen.getByText('¿Cómo hago una reserva?')).toBeInTheDocument();
    expect(
      screen.getByText(
        'Podés hacer una reserva desde la página de inicio seleccionando fechas y tipo de habitación.'
      )
    ).toBeInTheDocument();
  });

  test('simula click en un FAQItem', () => {
    const handleClick = vi.fn();
    FAQItem.mockImplementationOnce(({ question, answer }) => (
      <div data-testid="faq-item" onClick={handleClick}>
        <p>{question}</p>
        <p>{answer}</p>
      </div>
    ));
    render(<FAQList />);
    const item = screen.getAllByTestId('faq-item')[0];
    fireEvent.click(item);
    expect(handleClick).toHaveBeenCalled();
  });

  test('snapshot de FAQList', () => {
    const { container } = render(<FAQList />);
    expect(container).toMatchSnapshot();
  });
});
