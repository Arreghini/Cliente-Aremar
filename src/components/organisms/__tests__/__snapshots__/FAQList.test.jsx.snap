// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAQList > snapshot de FAQList 1`] = `
<div>
  <div
    class="space-y-4"
    data-testid="faq-list-container"
  >
    <div
      data-testid="faq-item"
    >
      <p>
        ¿Cómo hago una reserva?
      </p>
      <p>
        Podés hacer una reserva desde la página de inicio seleccionando fechas y tipo de habitación.
      </p>
    </div>
    <div
      data-testid="faq-item"
    >
      <p>
        ¿Se puede cancelar una reserva?
      </p>
      <p>
        Sí, podés cancelar desde tu perfil. Consultá nuestras políticas de cancelación.
      </p>
    </div>
    <div
      data-testid="faq-item"
    >
      <p>
        ¿Aceptan mascotas?
      </p>
      <p>
        En algunos departamentos sí, según disponibilidad. Consultá antes de reservar.
      </p>
    </div>
  </div>
</div>
`;
